pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -248,6 +250,8 @@ public void runSegmentMetrics() {
           (nReplicasIdealMax > 0) ? (nReplicasExternal * 100 / nReplicasIdealMax) : 100);
       _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.SEGMENTS_IN_ERROR_STATE,
           nErrors);
+      _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.PERCENT_OF_SEGMENTS,
+          (nSegments > 0) ? (100 - (nOffline/nSegments)*100) : 100);
```

Comment:
won't this always result in 100?

I think you need 

(nSegments > 0) ? (100 - (nOffline * 100 /nSegments)) : 100)